ThumbnailItem/ThumbnailItem.js

```diff
@@ -40,17 +92,22 @@ const ThumbnailItemBase = kind({
 		publicClassNames: true
 	},
 
-	render: ({children, css, src, ...rest}) => {
+	computed: {
+		className: ({type, selected, styler}) => styler.append({
+			roundThumbnail: type === 'styled',
+			selected
+		})
+	},
+
+	render: ({css, children, src, ...rest}) => {
 		return (
-			<Item
-				{...rest}
-			>
-				<slotBefore>
+			<Item {...rest} css={css}>
+				<div className={css.thumbnailContainer} slot="slotBefore">
```

Comment:
Sorry, I think the terminology I was using was confusing. Code is much clearer in this case.
```javascript
	render: ({css, children, src, slotBefore, ...rest}) => {
		return (
			<Item {...rest} css={css}>
				<slotBefore>
					<div className={css.thumbnailContainer}>
						<img
							className={css.thumbnail}
							src={src}
						/>
					</div>
					{slotBefore}
				</slotBefore>
				<div className={css.content}>
					{children}
				</div>
			</Item>
		);
	}
```

Here, I've extracted the `slotBefore` prop from `rest` –since we're replacing it by setting a `slotBefore`– and I've inserted the external/incoming `slotBefore` prop into our `<slotBefore>`.

Now, if your `.thumbnailContainer` node is still important, you can keep or drop it. I don't think it's necessary, but I don't know the history from the other conversation. (`agate/Image` includes a wrapper for `<img>`.)

Also, I'd suggest using the `agate/Image` component since it comes with some features you may want here. That may solve the disappearing/shift issue, provided you set a size.

Reply:
Oh, I got it now. Thanks for code sample!
I removed the `.thumbnailContainer` node and used `agate/Image` instead. This seems to solve the 'text shidting' issue as well.